onyo/commands/new.py

```diff
@@ -1,154 +1,296 @@
 
-    new_asset = Path(directory, create_filename(repo))
-    temp_asset = Path(repo.root, ".onyo/temp/", new_asset.name)
 
-    template = Path(template).resolve()
-    if template.is_file():
-        shutil.copyfile(template, temp_asset)
-    else:
-        log.error(f"'{template}' does not exist.")
-        sys.exit(1)
+def read_assets_from_tsv(tsv: str, template: Path, set_values: dict, repo: Repo) -> dict:
+    """
+    Read a tsv table with a header row and one row for each new asset to
+    create. Check the information (e.g. filenames correct and unique), and add
+    faux serial numbers to the name if needed.
 
-    if new_asset.exists():
-        log.error(f"'{new_asset}' asset already exists.")
-        sys.exit(1)
+    Returns a dictionary with paths and contents of assets on success, or
+    informs the user about missing or invalid information and exits with error.
+    """
+    new_assets = {}
+    existing_assets = repo.assets
+    row_count = 0
+
+    with open(tsv, 'r') as tsv_file:
+        # count and request the faux serial numbers needed
+        table = csv.DictReader(tsv_file, delimiter='\t')
+        faux_number = sum([1 for row in table if row['serial'] == 'faux'])
+        if faux_number > 0:
+            faux_serial_list = repo.generate_faux_serials(num=faux_number)
+
+        # iterate over the table from the beginning to read asset information.
+        tsv_file.seek(0)
+        table = csv.DictReader(tsv_file, delimiter='\t')
+        for row in table:
+            row_count += 1
+            new_path = ""
+            contents = {}
+
+            # error if any required field is empty
+            if not row['type'] or not row['make'] or not row['model'] or not row['serial'] or not row['directory']:
+                print("The fields 'type', 'make', 'model', 'serial' and 'directory' are required, but missing in line {row_count}.", file=sys.stderr)
+                sys.exit(1)
 
-    successful = True
-    if interactive:
-        editor = get_editor(repo)
-        successful = edit_asset(editor, temp_asset)
+            # set asset name and directory, add faux serial numbers if needed
+            if row['serial'] == 'faux':
+                row['serial'] = faux_serial_list.pop()
+            filename = f"{row['type']}_{row['make']}_{row['model']}.{row['serial']}"
+            directory = row['directory']
+            new_path = Path(repo.root, directory, filename).resolve()
+
+            # verify that the asset name is valid and unique in repo and table
+            valid_asset_destination(new_path, repo, existing_assets, new_assets)
+
+            # either a template is given in table, CLI, or onyo config
+            if "template" in row.keys():
+                if row['template'] == "":
+                    print(f"Template value missing in {row_count}", file=sys.stderr)
+                    sys.exit(1)
+                template = Path(repo.root, ".onyo/templates", row['template']).resolve()
```

Comment:
This feels like something that should be provided by the `repo` class.

Possibly a property `repo.templates` that is a set of Paths of all templates. And/or something that returns the Path object of the default template. The second I am uncertain about, but the first seems like a useful feature.